core/chainio/avs_reader.go

```diff
@@ -59,6 +70,73 @@ func (r *AvsReader) GetErc20Mock(tokenAddr gethcommon.Address) (*contractERC20Mo
 	return erc20Mock, nil
 }
 
-func (r *AvsReader) IsOperatorRegistered(address gethcommon.Address) (bool, error) {
+func (r *AvsReader) IsOperatorRegistered(address ethcommon.Address) (bool, error) {
 	return r.ChainReader.IsOperatorRegistered(&bind.CallOpts{}, address)
 }
+
+// Returns all the "NewBatchV3" logs that have not been responded starting from the given block number
+func (r *AvsReader) GetNotRespondedTasksFrom(fromBlock uint64) ([]servicemanager.ContractAlignedLayerServiceManagerNewBatchV3, error) {
```

Comment:
Should we create two functions `GetNotRespondedTasksFromV3` and `GetNotRespondedTasksFromV2` here and choose which one to call based on the operator version?

Reply:
I'm not sure if two functions are needed, but we should be able to handle both versions, right?